katalog/istio/README.md

```diff
@@ -17,169 +34,175 @@ Source: [https://istio.io/docs](https://istio.io/docs/concepts/what-is-istio/#wh
-  name: kiali
-  namespace: istio-system
-data:
-  config.yaml: |
-    istio_namespace: istio-system
-    deployment:
-      accessible_namespaces: ['**']
-    auth:
-      strategy: login
-    server:
-      port: 20001
-      web_root: /kiali
-    external_services:
-      tracing:
-        url:
-      grafana:
-        url:
-      prometheus:
-        url: http://prometheus-k8s.monitoring:9090
+bases:
+  - katalog/istio/init
+  - katalog/istio
+  - katalog/istio/telemetry/service-monitor
 ```
 
-Change the `external_services.prometheus.url` value with the correct prometheus server url.
+##### Note
+
+This package will deploy telemetry + service monitors. If telemetry it's already deployed, replace the telemetry base
+with the `katalog/istio/telemetry/service-monitor` base.
+
+
+### Egress Gateway
+
+This package deploys on top of the [minimal installation](#minimal-installation),
+[the egress gateway component](egress-gateway/README.md).
+
+Once deployed this package, don't forget to deploy the new Istio configuration based on the components installed:
+
+[Example kustomize installation file](../../examples/istio/minimal-and-egress/kustomization.yaml):
+
+```yaml
+namespace: istio-system
+
+bases:
+  - katalog/istio/init
+  - katalog/istio/minimal
+  - katalog/istio/egress-gateway
+  - katalog/istio/sidecar-injection/configuration/minimal-and-egress
+```
+
+### Citadel
+
+[Citadel](citadel/README.md) is the Istio component in charge of provide security features *(key and certificate management)*.
+If you are looking for mTLS, this is the component you should install.
+
+As the other components, this one should be installed on top of the minimal istio deployment.
+
+Example kustomize installation file:
+
+```yaml
+namespace: istio-system
+
+bases:
+  - katalog/istio/init
+  - katalog/istio/minimal
+  - katalog/istio/citadel
+  - katalog/istio/egress-gateway
+  - katalog/istio/sidecar-injection/configuration/citadel-and-egress
+```
+
+In this example you can see istio + citadel + egress gateway installed.
+
+##### Note
+
+Be sure to deploy the correct `sidecar-injection/configuration` kustomize project. In this example the correct
+configuration is the one with [citadel + egress](sidecar-injection/configuration/citadel-and-egress).
+
+
+### Sidecar injector
+
+[Sidecar injector](sidecar-injection/README.md) is the component responsible of injecting the envoy proxy on every
+required pod. This is the alternative of executing `istioctl inject` comamnds.
+
+Sidecar injector package requires [citadel](citadel/README.md) installed.
+
+Example kustomize installation file:
+
+```yaml
+namespace: istio-system
+
+bases:
+  - katalog/istio/init
+  - katalog/istio/minimal
+  - katalog/istio/citadel
+  - katalog/istio/egress-gateway
+  - katalog/istio/sidecar-injection
+  - katalog/istio/sidecar-injection/configuration/sidecar-injection-and-egress
+```
+
+In this example you can see istio + citadel + egress gateway and the sidecar injector installed.
```

Comment:
"With the previous example snippet you will get `istio` with `citadel`, `egress gateway` and `sidecar injector` modules installed."

Reply:
Solved!